pkg/pgsql/server/query_machine.go

```diff
@@ -336,7 +336,7 @@ func (s *session) inferParamAndResultCols(statement string) ([]*schema.Column, [
 			return nil, nil, err
 		}
 		for _, c := range cols {
-			resCols = append(resCols, &schema.Column{Name: c.Selector(), Type: c.Type})
+			resCols = append(resCols, &schema.Column{Name: c.Column, Type: c.Type})
```

Comment:
I know the database/sql descriptor is the descriptor that is exposed to the client.
But for this one I don't really know what its usage is

Reply:
if just the column name is used then queries like "select t1.id, t2.id ..." could lead to wrong interpretations, that's why full descriptor "db.table.col" is used instead... but the final layer may provide a way to just fetch the column name, or some way of dealing with such conflicts...